src/View/ViewLayerData.php

```diff
@@ -0,0 +1,254 @@
+<?php
+
+namespace SilverStripe\View;
+
+use BadMethodCallException;
+use InvalidArgumentException;
+use IteratorAggregate;
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\Model\ModelData;
+use SilverStripe\Model\ModelDataCustomised;
+use SilverStripe\ORM\FieldType\DBClassName;
+use Stringable;
+use Traversable;
+
+class ViewLayerData implements IteratorAggregate, Stringable
```

Comment:
```suggestion
class TemplateData implements IteratorAggregate, Stringable
```
Is `TemplateData` an accurate name?  ViewLayerData is kind of weird and makes me just want to call it ViewableData, which we shouldn't do :-)  We don't call things ModelLayerData so seems like we shouldn't include the word 'Layer' here either

Reply:
It can be used anywhere in the view layer, it doesn't have to be restricted to templates.

> We don't call things ModelLayerData so seems like we shouldn't include the word 'Layer' here either

That's because it is the data that represents a model, it's not data _for the model layer_.
This, on the other hand, is data _for the view layer_, it doesn't _represent_ the view or the entirety of the view layer.

All of that said, I really don't feel strongly about it, so if you want me to call it `TemplateData` I can do that. Just know that makes it harder to justify using it for other areas in the view layer later if we want to.

Reply:
"View layer" is not a term that I've seen previously used. What specifically makes up the "view layer"?

Reply:
It's often called the "presentation layer" which you've probably heard Jackson reference a few times. It's basically any part of "presenting" data to the end user. So we could use this class to wrap data inside GridField in order to generate the presentation of that field, for example.

Reply:
Could you please add a class docblock explaining what this class does

Reply:
Yes! Thank you for catching that. It's way too easy to forget.